Fix BC date off-by-one bug

noinkling opened the pull request:
Fixes #5 

Also fixes BC `date` type output, which currently isn't parsing at all. e.g. `1234-01-01 BC`

Related: brianc/node-postgres#1864

bendrucker commented on index.js:
```diff
@@ -43,7 +45,7 @@ module.exports = function parseDate (isoDate) {
   }
 
-  if (isFirstCentury) {
+  if (yearIsBetween0And99) {
```
Would rather remove this var and just move the check here, since it's not reused

bendrucker commented on index.js:
```diff
@@ -52,7 +54,13 @@ module.exports = function parseDate (isoDate) {
+
   var year = parseInt(matches[1], 10)
+  var isBC = !!matches[4]
+  if (isBC) {
```
isBC is calculated elsewhere with a second regex. Fine with using the match from the main one, but please update the other fn to match and remove `BC`.

noinkling commented:
I updated the BC detection as requested.

Instead of inlining the "0 to 99" check I fixed another edge case due to `setUTCFullYear` being used for local times (as well as timezone qualified ones), when it should be using `setFullYear`. For example, in my current timezone:

```js
> var date1 = new Date("1976-01-01T01:15:30+12:00")
> var date2 = new Date(date1)
> date1.setUTCFullYear(76)
-59737257870000
> date2.setFullYear(76)
-59768875414000
> date1.toISOString()
"0076-12-31T13:15:30.000Z"
> date2.toISOString()
"0075-12-31T14:36:26.000Z"
```

I didn't add a test because it's local timezone dependent and I don't know how to control that (at least without a more elaborate setup).

noinkling commented:
...And fixed one more bug, with a test case for this one. Hopefully that's all.

bendrucker commented:
Thanks!
